crates/sui-indexer/src/main.rs

```diff
@@ -27,22 +28,44 @@ async fn main() -> Result<(), IndexerError> {
 
     let indexer_config = IndexerConfig::parse();
     retry(ExponentialBackoff::default(), || async {
-        let rpc_client = new_rpc_client(RPC_CLIENT_URL.into()).await?;
+        let rpc_client = new_rpc_client(indexer_config.rpc_client_url.clone()).await?;
         let pg_connection_pool = new_pg_connection_pool(indexer_config.db_url.clone()).await?;
         // NOTE: Each handler is responsible for one type of data from nodes,like transactions and events;
         // Handler orchestrator runs these handlers in parallel and manage them upon errors etc.
         let handler_rpc_client = rpc_client.clone();
         let handler_pg_pool = pg_connection_pool.clone();
+
+        let registry_service = start_prometheus_server(
+            // NOTE: this parses the input host addr and port number for socket addr,
+            // so unwrap() is safe here.
+            format!(
+                "{}:{}",
+                indexer_config.client_metric_host, indexer_config.client_metric_port
```

Comment:
FYI you can test the metrics out locally by following the summary in https://github.com/MystenLabs/sui/pull/6871